src/main/java/com/onarandombox/multiverseinventories/share/Sharables.java

```diff
@@ -582,6 +590,149 @@ public boolean updatePlayer(Player player, PlayerProfile profile) {
+                        }
+                        for (String criteria: intersection) {
+                            processedAdvancements.add(criteria);
+                            completedAdvancements.remove(criteria);
+                            player.getAdvancementProgress(advancement).awardCriteria(criteria);
+                        }
+
+                        // here's the idea: revoke all (criteria \ completedAdvancements), grant (criteria intersect completedAdvancements)
+                        // also, we don't need to grant/revoke anything in processedAdvancements since they've already been granted/revoked!
+                    }
+                    if (announceAdvancements) player.getWorld().setGameRule(GameRule.ANNOUNCE_ADVANCEMENTS, true);
+                    return advancements != null;
+                }
+            }).defaultSerializer(new ProfileEntry(false, "advancements")).altName("achievements").build();
+
+    /**
+     * Sharing Statistics.
+     */
+    public static final Sharable<Map> GAME_STATISTICS = new Sharable.Builder<Map>("game_statistics", Map.class,
+            new SharableHandler<Map>() {
+                @Override
+                public void updateProfile(PlayerProfile profile, Player player) {
+                    Map<String, Integer> playerStats = new HashMap<>();
+                    for (Statistic stat: Statistic.values()) {
+                        if (stat.getType() == Statistic.Type.UNTYPED) {
+                            int val = player.getStatistic(stat);
+                            // no need to save values of 0, that's the default!
+                            if (val != 0) playerStats.put(stat.name(), val);
+                        }
+                    }
+                    profile.set(GAME_STATISTICS, playerStats);
+                }
+
+                @Override
+                public boolean updatePlayer(Player player, PlayerProfile profile) {
+                    Map<String, Integer> playerStats = profile.get(GAME_STATISTICS);
+                    for (Statistic stat : Statistic.values()) {
+                        if (stat.getType() == Statistic.Type.UNTYPED) player.setStatistic(stat, 0);
+                    }
+                    if (playerStats == null) {
+                        return false;
+                    }
+                    for (String stringStat : playerStats.keySet()) {
+                        Statistic stat = Statistic.valueOf(stringStat);
+                        if (stat.getType() == Statistic.Type.UNTYPED) player.setStatistic(stat, playerStats.get(stat.name()));
+                    }
+                    return true;
+                }
+            }).defaultSerializer(new ProfileEntry(false, "game_statistics")).altName("game_stats").build();
+
+    /**
+     * Sharing Recipes.
+     */
+    public static final Sharable<Map> RECIPES = new Sharable.Builder<Map>("recipes", Map.class,
```

Comment:
Is there is reason why generics datatype used is `Map` and not `Map<String, List<String>>`?

Reply:
It doesn't compile when I use `Map<String, List<String>>`, I can't remember if I tried to figure out why or if I just went with it.